contrib/mesos/pkg/executor/executor.go

```diff
@@ -1044,3 +1002,220 @@ func executorInfoToAnnotations(ei *mesos.ExecutorInfo) (annotations map[string]s
 
 	return
 }
+
+// implements txTaskOperator, uses the executor state lock to handle concurrent invocation
+func (k *Executor) doForTask(taskId string, f func(*kuberTask, *api.Pod) bool) (changed bool, err error) {
+	k.lock.Lock()
+	defer k.lock.Unlock()
+
+	// exclude tasks which are already deleted from our task registry
+	task := k.tasks[taskId]
```

Comment:
i thought about doing that in this PR. was unhappy about the scope creep that I was already experiencing. i like the idea of the abstraction. will consider for this PR, but might push to a follow-up PR